modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotMetadata.java

```diff
@@ -221,6 +253,15 @@ private void readObject(java.io.ObjectInputStream s) throws IOException, ClassNo
 
             locParts.put(grpId, parts);
         }
+
+        if (hasComprGrps) {
+            int sz = s.readInt();
```

Comment:
U.readSet / U.writeCollection can be reused

Reply:
Yep, it is probably better.